Lab2_Code.m

```diff
@@ -0,0 +1,349 @@
+%% Lab 2
+
+%% Problem 1
+% In this section, we see gamma-ray sources and a background of cosmis-rays
+% using Fermi telescope. I selected my average cosmic-ray background value as 
+% X = 8 and my average number of gamma-rays emmitted as Y = 16. 
+
+%% Section 1A
+% Shows how the probability distribution changes as we integrate (sum) for
+% more days by convolving the background with itself N times. We notice how
+% the mean shifts, the width becomes thinner and the amplitude at the mean 
+% decreases as we integrate for more days. Also, the distribution is still 
+% a Poisson distribution. 
+pdist = makedist("Poisson",'lambda',8);
+pbg = pdf(pdist,0:20);
+
+% Sum over 2 days
+pconv = multiconv(pbg,2);
+px = 0:length(pconv)-1;
+stairs(px, pconv); hold on;
+
+% Sum over 3 days
+pconv = multiconv(pbg,3);
+px = 0:length(pconv)-1;
+stairs(px, pconv); hold on;
+
+% Sum over 4 days
+pconv = multiconv(pbg,4);
+px = 0:length(pconv)-1;
+stairs(px, pconv); hold on;
+
+title("Probability Distribution of Background Summed Over N Days");
+xlabel("Number of Cosmic-Ray Background"); ylabel("Probability");
+legend("N = 2","N = 3","N = 4"); hold off; figure;
+
+% Below, we put these graphs from above in semilog plot. 
+
+% Sum over 2 days in log scale
+pconv = multiconv(pbg,2);
+px = 0:length(pconv)-1;
+stairs(px, pconv);
+set(gca,'YScale','log'); hold on;
+
+% Sum over 3 days in log scale
+pconv = multiconv(pbg,3);
+px = 0:length(pconv)-1;
+stairs(px, pconv); 
+set(gca,'YScale','log'); hold on;
+
+% Sum over 4 days in log scale
+pconv = multiconv(pbg,4);
+px = 0:length(pconv)-1;
+stairs(px, pconv); 
+set(gca,'YScale','log'); hold on;
+
+xlim([0 40]);
+title("Probability Distribution of Background Summed Over N Days");
+xlabel("Number of Cosmic-Ray Background"); 
+ylabel("Probability in Log Scale");
+legend("N = 2","N = 3","N = 4",'Location','Southeast'); hold off; figure;
+
+%% Section 1B
+% Shows how the summed probability distribution is still a Poisson
+% distribution after 5 days. 
+
+% Sum over 5 days in log scale
+pconv = multiconv(pbg,5);
+px = 0:length(pconv)-1;
+subplot(2,1,2);
+stairs(px, pconv); 
+title("Probability Distribution After 5 Days in Log Scale"); 
+xlabel("Number of Cosmic-Ray Background"); 
+ylabel("Probability in Log Scale"); 
+set(gca,'YScale','log')
+
+% Sum over 5 days
+subplot(2,1,1)
+stairs(px, pconv); 
+title("Probability Distribution After 5 Days");
+xlabel("Number of Cosmic-Ray Background"); 
+ylabel("Probability"); figure;
+
+% Here, we see how the summed probability distribution is still a Poisson
+% distribution. From a mathematical perspective, this makes sense because
+% the distribution does not have a small enough step size to approximate a
+% Gaussian distribution over 5 days. From a conceptual perspective, this
+% makes sense because we cannot have "negative days" which is why we don't
```

Comment:
did you mean to say negative counts? the x axis of the dist is not time but number of events in a given time